### ISSUE TITLE
'BestCommonType error' returns wrong error range inside class methods

### ISSUE THREAD

**Issue by simonvizzini** [Microsoft/TypeScript#5732]
I'm working on a JS project with ES6 syntax and today I tried out the latest VS Code release (0.10.1). Today I've also learned that the IntelliSense support in VS Code for JS is coming from TS, so I've been told to report it here (https://github.com/Microsoft/vscode/issues/230)

This is the result when there is a "BestCommonType" error inside a class method:
![image](https://cloud.githubusercontent.com/assets/10862380/11303691/72b9d93e-8fa4-11e5-8a78-c3faf175257b.png)

And for plain functions it looks like this:
![image](https://cloud.githubusercontent.com/assets/10862380/11303731/b53bf99a-8fa4-11e5-96b2-24afa23efa31.png)

**Comment by RyanCavanaugh** [comment]
Text transcription of a repro:
```ts
class Foo {
	doSomething(opts) {
		if(true) {
			return 42;
		}
		
		return 'nope';
	}
}
```

**Comment by DanielRosenwasser** [comment]
Thanks @reavowed!
